dbt/include/athena/macros/materializations/models/table/table.sql

```diff
@@ -133,20 +133,27 @@
         {%- endif -%}
 
         {%- set old_relation_table_type = adapter.get_glue_table_type(old_relation) -%}
+        -- we cannot use old_bkp_relation, because it returns None if the relation doesn't exist
+        -- we need to create a python object via the make_temp_relation instead
+        {%- set old_relation_bkp = make_temp_relation(old_relation, '__bkp') -%}
 
-        {%- if old_relation_table_type == 'iceberg' -%}
-          {{ rename_relation(old_relation, old_bkp_relation) }}
+        {%- if old_relation_table_type.value == 'iceberg_table' -%}
+          {{ rename_relation(old_relation, old_relation_bkp) }}
```

Comment:
has it been considered, that the rename might have never occurred, as the old relation value was not being correctly compared, and so, it was always None?

Reply:
relation type cannot be None. here you are in a part of the code where old_relation definitely exists AND we need to get the type of relation.
If the old relation is an external table == hive, we don't rename, because hive doesn't support renaming, but if the old relation is iceberg, we can simply rename the old relation to `old_relation_bkp`.

Reply:
@octiva indeed, what you raised is actually a good idea to check.
the renaming happen only from old target being iceberg, for hive to iceberg old_relation_bkp doesn't exists for sure.

Reply:
@octiva I should have addressed your concern here adding a check if old_relation exists, only then calling get_glue_table_type